detector/cve/cve20258088/cve20258088.go

```diff
@@ -0,0 +1,331 @@
+
+					affected := []osvschema.Affected{{
+						Package: osvschema.Package{
+							Name:      pkg.Name,
+							Ecosystem: "generic",
+						},
+						Ranges: []osvschema.Range{{
+							Type: "SEMVER",
+							Events: []osvschema.Event{{
+								Introduced: "0",
+							}, {
+								Fixed: "7.13",
+							}},
+						}},
+						Versions: []string{normalizedVersion},
+					}}
+
+					vuln := &inventory.PackageVuln{
+						Package: pkg,
+						Vulnerability: osvschema.Vulnerability{
+							ID:       "CVE-2025-8088",
+							Summary:  "WinRAR vulnerable version detected (installed package)",
+							Details:  fmt.Sprintf("Installed WinRAR %s detected via package index", normalizedVersion),
+							Affected: affected,
+						},
+					}
+
+					findings = append(findings, vuln)
+				}
+			}
+		}
+	}
+
+	// === Phase 2: Filesystem scan (existing logic) ===
+	err := fs.WalkDir(scanRoot.FS, scanRoot.Path, func(path string, d fs.DirEntry, err error) error {
+		if err != nil || d.IsDir() {
+			log.Debugf("cve20258088: Skipping directory or directory entry: %s : error %s", path, err)
+			return nil
+		}
+
+		ext := strings.ToLower(filepath.Ext(d.Name()))
+		if ext != ".exe" && ext != ".dll" {
+			return nil
+		}
+
+		confidence := 0
+		var component string
+
+		// Indicator 1: Filename match
+		if peRegex.MatchString(d.Name()) {
+			confidence += 2
+			matches := peRegex.FindStringSubmatch(d.Name())
+			if len(matches) > 1 {
+				component = matches[1]
+			}
+		}
+
+		// Indicator 2: PE resource analysis - extract version info from PE resources
+		// Now check PE resources for ALL .exe/.dll files, not just regex matches
+		version, prodName := extractPEVersion(path)
+		if prodName != "" {
+			// Check if the product name indicates WinRAR/RAR/UnRAR
+			prodNameLower := strings.ToLower(prodName)
+			if strings.Contains(prodNameLower, "winrar") || strings.Contains(prodNameLower, "rar") || strings.Contains(prodNameLower, "unrar") {
+				confidence += 2
+				if component == "" {
+					component = prodName
+				}
+			}
+		}
+
+		// Skip if we have no indication this is a WinRAR-related file
+		if confidence == 0 {
+			log.Debugf("cve20258088: Skipping file %s, no WinRAR indicators found", d.Name())
+			return nil
+		}
+
+		// Only report if confidence is high enough
+		if confidence >= 2 && version != "" {
+			normalizedVersion := normalizeVersion(version)
+
+			if isAffectedVersion(normalizedVersion) {
+				log.Infof("Vulnerable WinRAR package found(Delete them): %s, Package: %s, Version: %s", path, prodName, normalizedVersion)
+				pkg := &extractor.Package{
+					Name:         component,
+					Version:      normalizedVersion,
+					PURLType:     "generic",
+					Locations:    []string{path},
+					SourceCode:   &extractor.SourceCodeIdentifier{},
+					LayerDetails: &extractor.LayerDetails{},
+					Metadata:     nil,
+				}
+
+				affected := []osvschema.Affected{{
```

Comment:
Same as for step 1, pls move the vuln creation logic into a helper func and use it both here and in DetectedFinding (with potentially different databaseSpecific data)